### PR TITLE
refactor: switch to thin LTO for faster builds

### PR DESCRIPTION
**Pull request by westonpace**

### CODE REVIEW

**Review by chebbyChefNEQ — APPROVED** [review]

File: rust/.cargo/config.toml [hunk 1]
```diff
@@ -1,5 +1,13 @@
...
-lto = true
+lto = "thin"
+codegen-units = 1
+
+[profile.release-with-debug]
+inherits = "release"
+debug = true
+# Prioritize compile time over runtime performance
+codegen-units = 16
+lto = "off"
```
Comment (chebbyChefNEQ) [inline rust/.cargo/config.toml:10]: Maybe we should keep this at thin? Just to avoid CI and local producing different results?

Comment (westonpace) [reply rust/.cargo/config.toml]: Done.  Good idea.  Do we still want the python builds diverging?  Or should we use thin for both there as well?

Comment (chebbyChefNEQ) [reply rust/.cargo/config.toml]: yeah, i guess we should do the same for python.

Comment (westonpace) [reply rust/.cargo/config.toml]: Done.

**Review by chebbyChefNEQ — APPROVED** [review]

**Pull request merged**